index_configuration_tool/tests/test_main.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import pickle
+import random
+import unittest
+from os.path import dirname
+from typing import Optional
+
+import main
+
+# Constants
+LOGSTASH_EXPECTED_OUTPUT_FILE = dirname(__file__) + "/expected_parse_output.pickle"
+TEST_KEY = "test_key"
+BASE_DATA = {
+    TEST_KEY: [("invalid_plugin1", None), ("invalid_plugin2", {})]
+}
+
+
+# Utility method to create a test plugin config
+def create_plugin_config(host_list: list[str],
+                         ssl: Optional[bool] = None,
+                         user: Optional[str] = None,
+                         password: Optional[str] = None) -> dict:
+    config = dict()
+    if len(host_list) == 1:
+        config["hosts"] = host_list[0]
+    else:
+        config["hosts"] = host_list
+    if ssl:
+        config["ssl"] = ssl
+    if user:
+        config["user"] = user
+    if password:
+        config["password"] = password
+    return config
+
+
+# Utility method to creat a test config section
+def create_config_section(plugin_config: dict) -> dict:
+    valid_plugin = (random.choice(main.SUPPORTED_ENDPOINTS), plugin_config)
+    config_section = copy.deepcopy(BASE_DATA)
+    config_section[TEST_KEY].append(valid_plugin)
+    return config_section
+
+
+class TestMain(unittest.TestCase):
+    # Run before each test
+    def setUp(self) -> None:
+        with open(LOGSTASH_EXPECTED_OUTPUT_FILE, "rb") as f:
+            self.loaded_logstash_config = pickle.load(f)
+
+    def test_get_auth(self):
+        # Empty input should produce empty auth tuple
+        self.assertFalse(main.get_auth(dict()))
+        base_data = {"user": "user", "password": "password"}
+        # Password without user should return empty auth tuple
+        test_data = dict(base_data)
+        del test_data["user"]
+        self.assertFalse(main.get_auth(test_data))
+        # User without password should return empty auth tuple
+        test_data = dict(base_data)
+        del test_data["password"]
+        self.assertFalse(main.get_auth(test_data))
+        # Test valid input
+        result = main.get_auth(base_data)
+        assert type(result) is tuple
+        self.assertEqual("user", result[0])
+        self.assertEqual("password", result[1])
+
+    def test_get_endpoint_info(self):
+        # Simple base case
+        test_config = create_plugin_config(["test"])
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL enabled
+        test_config = create_plugin_config(["test"], True)
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("https://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL disabled, invalid auth config
+        test_config = create_plugin_config(["test"], False, "test")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
```

Comment:
what is the actual value here? is it None or False (or something else)?

Reply:
The expected value here is `None`. Fixed to be strongly-typed, and added deeper checks for the with-auth cases